app/src/lib/components/PowerTable.svelte

```diff
@@ -839,12 +840,24 @@ export function addAction(e: Event) {
 
 export function deleteAction(e: Event) {
     closeMenu(e);
+    
+    let deletedRows = [];
+    let remainingRows = [];
+    for (const row of data) {
+        if (row[checkboxKey]) {
+            deletedRows.push(row);
+        } else { 
+            remainingRows.push(row);
+        }
+    }
 
-    data = data.filter(row => {
-        return !row[checkboxKey];
-    });
-
+    data = remainingRows;
```

Comment:
Thank you for your contribution! Adding the callbacks is a great idea.
As for these lines, I think it would be more efficient to push the deleted rows into `deletedRows` array from within the filter method. This way we won't need to copy all the remaining rows to a new array.

Reply:
Ah, fixed!